### ISSUE TITLE
Commandline option to write the whole rulesets including default properties.

### ISSUE THREAD

**Issue by PowerStat** [checkstyle/checkstyle#12377]
I already propsed this for pmd, but would like to see it for ChecklStyle in the same way:

**Is your feature request related to a problem? Please describe.**
From time to time (not regularly) I update my CheckStyle configs. And then it is hard to find out what I have changed in my config and what you have changed.

**Describe the solution you'd like**
A commandline option to write the whole known rulesets including the default properties into a checkstyle-default.xml file.
This will allow me to have a simple diff between my configurations and your configuration with all its defaults.
This requires that you write out your configuration in a defined way, like first by programming language, then by section and then alphabetically. The properties should also written out alphabetically.
Also this might allow to automatically update the rules in a CI/CD context.

**Describe alternatives you've considered**
Verifying my config with your website rules section takes to much time and is to error prone!

**Additional context**
None

**Comment by rnveach** [comment]
> I already propsed this for pmd

PMD Issue at https://github.com/pmd/pmd/issues/4187

**Comment by romani** [comment]
Why you need all Checks in one file, nobody use all Checks at same time, some of them conflicts with each other, some of them should be used in very narrow code bases.

What if we do this only for one Check ?

https://checkstyle.org/cmdline.html

`java -jar checkstyle-10.4-all.jar --print-default-config=LineLengh`

Output:
```
<module name="LineLength">
  <property name="fileExtensions" value="all files"/>
  <property name="ignorePattern" value="^(package|import) .*""/>
  <property name="max" value="80"/>
</module>
```

**Comment by nrmancuso** [comment]
> What if we do this only for one Check ?

It would be better to allow a list of checks to be passed to `--print-default-config`, so that users can do something like the following:
```bash

java -jar checkstyle-xx-all.jar --print-default-config=$(cat config.xml | grep "module name" | cut -d'"' -f2 | sed -z 's/\n/,/g;s/,$/\n/')
```

Where the command in the subshell would print a comma separated list of all modules in your config, like so:
```
$ cat config.xml | grep "module name" | cut -d'"' -f2 | sed -z 's/\n/,/g;s/,$/\n/'
Checker,TreeWalker,LineLength,RightCurly,ParameterName,...

```

This list could be used to generate a config that matches an existing config in your project, but all with default values.

**Comment by romani** [comment]
@nrmancuso , you are experienced in shell, there numerous engineers who works on windows. Where all your magic is not available

**Comment by nrmancuso** [comment]
@romani see edit form above:

>Or, we could just create some way to supply a config file and have a similar one with default values returned?

What do you think about this? Example:
```
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="ParameterName">
            <property name="format" value="^[a-z][_a-zA-Z0-9]+$"/>
        </module>
    </module>
</module>

$ java -jar checkstyle-xx-all.jar --print-defaults config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="ParameterName">
            <property name="format" value="^[a-z][a-zA-Z0-9]*$"/>
        </module>
    </module>
</module>


```
?

**Comment by romani** [comment]
Good idea, much better, if we preserve formatting as is and only change value of property that is mentioned source config.

In other cases it would be hard to compare.

**Comment by PowerStat** [comment]
@romani I require all checks ithin one file, because I would like to compare it at once - keep in mind that in my config file I use comments  to switch off rules I don't need - but I am still interested in them for the case that they get new properties. Exporting single rules is completly useless for me. Also when talking about diff I mean Linux and Windows - for Windows you could use WinMerge which does the job perfectly.

**Comment by romani** [comment]
You asking too much, we will not automate what you need in your CI/CD.
But

We can print by CLI default value of property in Check. To let you do extra script on your side and generate file in way you want.

`--print-default-property-value=MagicNumber.ignoreNumbers`

Will print `-1, 0, 1, 2` .

FYI: https://checkstyle.org/config_coding.html#MagicNumber_Properties
